allen-cell-animated/timelapse-colorizer
# feature: Feature glossary

Problem
=======
Closes #198, "glossary for feature descriptions."

Implements a feature glossary that slides out on button click! Dataset authors can provide a "description" field for each feature, and the contents of the loaded data will appear in the glossary.

*Estimated size: small, 15 minutes (turn off whitespace)*

Solution
========
- Adds the `GlossaryPanel` component, which renders the list of features in an Ant `Drawer`.
- Adds support for the `descriptions` feature metadata added in `colorizer-data v.1.4.2`.
- Bonus: refactors the `LandingPage` to use the `ExternalLink` convenience component.

## Type of change
* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Open the preview link, which has a dataset I manually added feature descriptions to: https://allen-cell-animated.github.io/timelapse-colorizer/pr-preview/pr-435/viewer?collection=https%3A%2F%2Fdev-aics-dtp-001.int.allencell.org%2Fdan-data%2Fcolorizer%2Fdata%2Ftest%2Fparquet%2Fcollection.json&dataset=Large+Parquet
2. Click the book icon next to the features dropdown. The panel should open and be scrollable.

Screenshots (optional):
-----------------------
![image](https://github.com/user-attachments/assets/f18caa97-1bf2-4bae-a4ba-cf728f52f260)

**Video talkthrough (🔊):**

https://github.com/user-attachments/assets/e0fee0b3-e2a1-4214-a4c4-d9b2ef3bac0d
